pytorch_lightning/trainer/training_loop.py

```diff
@@ -333,9 +333,15 @@ def train(self):
         self.reset_train_dataloader(model)
         self.reset_val_dataloader(model)
 
-        # Train begin callbacks
-        model.on_train_start()
-        self.on_train_start()
+        # Train start events
+        with self.profiler.profile('on_train_start'):
+            # callbacks
+            self.on_train_start()
+            # initialize early stop callback
+            if self.early_stop_callback is not None:
+                self.early_stop_callback.on_train_start(self, self.get_model())
```

Comment:
just curious what `early_stop_callback` can do just after the be, maybe for resumed training?

Reply:
sorry i'm not following, can you elaborate?